### ISSUE TITLE
Rule proposal: `multi-word-component-names`

### ISSUE THREAD

**Issue by michalsnik** [vuejs/eslint-plugin-vue#245]
We don't have a rule for checking this and it's already in style guide, so it might be a good idea to add it here too :)

More informations here:
https://vuejs.org/v2/style-guide/#Multi-word-component-names-essential

**Comment by chrisvfritz** [comment]
Note that for `.vue` files, the file basename without extension should be treated as the name if a `name` option is not present.

**Comment by michalsnik** [comment]
But I guess we also want to report this case @chrisvfritz, aren't we?
```
// Header.vue
{
  name: 'TheHeader',
  ...
}
```

**Comment by chrisvfritz** [comment]
@michalsnik Yeah, if a `name` option is present, the filename should be ignored in this case.

**Comment by armano2** [comment]
note:
`executeOnVueComponents` is not triggered when 2nd parameter of `Vue.component` is not an object

**Comment by samboylett** [comment]
Any news on this?

**Comment by theoptz** [comment]
any updates?

**Comment by gkatsanos** [comment]
So we prefer to have to MAKE UP extra words for possibly hundreds of components instead of not using reserved words/html standard elements? This rule is quite.. exotic if I may.

**Comment by manniL** [comment]
This is not ideal for a `pages` folder which mirrors the URL structure (like nuxt has). Files like `index.vue` or `about.vue` should be fine 🙈

**Comment by FloEdelmann** [comment]
@gkatsanos Please see the reasoning in the Vue Style Guide: https://v3.vuejs.org/style-guide/#multi-word-component-names-essential

@manniL You can disable this rule selectively for these cases: https://eslint.org/docs/user-guide/configuring/configuration-files#cascading-and-hierarchy or https://eslint.org/docs/user-guide/configuring/configuration-files#configuration-based-on-glob-patterns.
